### ISSUE TITLE
Read Jurafsky & Martin chapter 7

### ISSUE THREAD

**Issue by rockladyeagles** [vgcagle/DARC#3]
When you're done, write your takeaways from the chapter in a comment on this issue, then remove yourself as an Assignee. Last one to do so closes the issue.

**Comment by harmonyrose** [comment]
A neural network takes a vector of inputs and produces one output. The hidden layer is formed of neural units that take the weighted sum of its inputs and apply a non-linear activation function to get the output. In feedforward networks, computation proceeds iteratively from one layer of units to the next. In recurrent neural networks, connections can cycle back, allowing us to deal with sequential data, holding knowledge about the past. Forward inference (otherwise known as decoding), given an input, runs a forward pass on a neural network to produce a probability distribution over possible outputs. This can be used for language modeling, which is predicting upcoming words from prior word context

**Comment by rpersing** [comment]
Compared to n-gram models, neural language models can handle much longer histories, can generalize better over contexts of similar words, and are more accurate at word prediction. 

A feedforward neural LM approximates the probability of a word given the entire prior context. 

Neural language models represent words in this prior context by their embeddings, rather than just by their word identity as used in n-gram language models.

**Comment by vgcagle** [comment]
The computation for a neural net only has three steps: multiplying the weight matrix by the input vector x, adding the bias vector b, and applying the activation function g (such as the sigmoid, tanh, or ReLU activation function ). 

There are no cycles in a feed-forward neural network. 

To compute the gradients of the loss function for a network, error backpropagation, backward differentiation on a computation graph, is used.
